### PR TITLE
FW-1953 Pre-Chat lead form - Dropdown list on Web

### PR DESCRIPTION
**Pull request by Ojashwa**
### What do you want to achieve?

- Create a dropdown list for a pre-chat lead form field.

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [ ] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- Load installation script with preLeadCollection as suggested in the below image.

- Check a dropdown for that field will appear in the chat widget.
---
image
### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 1]
```diff
@@ -2936,15 +2934,32 @@ var userOverride = {
...
-                        'aria-label',
-                        preLeadCollection.field
-                    );
+                    if (
+                        preLeadCollection.element == 'select' &&
+                        preLeadCollection.options &&
+                        mckMessageService.checkArray(preLeadCollection.options)
+                    ) {
+                        var dropDownOptions = preLeadCollection.options;
+                        var setDropDown = _this.createSelectFieldDropdown(
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:2944]: dropDownOptionsdropDownOptions and setDropDown variable is not required, as it is used only once, directly use 
kmChatInput = _this.createSelectFieldDropdown(preLeadCollection.options, kmChatInput);

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 2]
```diff
@@ -2959,6 +2974,37 @@ var userOverride = {
...
             };
+
+            _this.createSelectFieldDropdown = function (
+                options,
+                selectElement
+            ) {
+                var dropDownOption = document.createElement('option');
+                dropDownOption.setAttribute('value', '');
+                dropDownOption.innerHTML =
+                    'Please select ' +
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:2985]: use MCK_LABELS for 'Please select' so that customers can configure it if they wish to change the text

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/labels/default-labels.js [hunk 3]
```diff
@@ -97,6 +97,7 @@ Kommunicate.defaultLabels = {
     'conversation.rated': 'rated the conversation',
     'offline.msg':
         'Uh oh! No internet connection. Please check your network settings and try again.',
+    'select.option': 'Please select',
```
Comment (devashishmamgain) [inline webplugin/js/app/labels/default-labels.js:100]: See if there is any labels for pre chat lead collection, put it under them

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Review by devashishmamgain — CHANGES_REQUESTED** [review]
Verify naming convention used in labels

**Review by devashishmamgain — APPROVED** [review]

**Review by Prochnost55 — APPROVED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 4]
```diff
@@ -2959,6 +2972,37 @@ var userOverride = {
...
+                        .getAttribute('name')
+                        .toLowerCase()
+                        .split('-')[1];
+                selectElement.append(dropDownOption);
+                options.forEach((element) => {
+                    if (kommunicateCommons.isObject(element)) {
+                        dropDownOption = document.createElement('option');
+                        dropDownOption.value = element.value;
+                        dropDownOption.innerHTML =
+                            element.value.charAt(0).toUpperCase() +
```
Comment (Prochnost55) [inline webplugin/js/app/mck-sidebox-1.0.js:2994]: remove arrow function

**Review by Prochnost55 — CHANGES_REQUESTED** [review]

**Review by devashishmamgain — DISMISSED** [review]

**Review by Prochnost55 — DISMISSED** [review]

**Comment by Prochnost55** [comment]
Resolve the merge conflict

**Review by devashishmamgain — APPROVED** [review]

**Review by Prochnost55 — DISMISSED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 5]
```diff
@@ -2936,17 +2934,32 @@ var userOverride = {
...
+                            preLeadCollection.placeholder || ''
+                        );
+                        kmChatInput.setAttribute(
+                            'aria-label',
+                            preLeadCollection.field
+                        );
+                    }
                     $applozic(kmChatInputDiv).append(kmChatInput);
+                    $applozic('.km-last-child').append(kmChatInputDiv);
                 }
```
Comment (Prochnost55) [inline webplugin/js/app/mck-sidebox-1.0.js:2963]: avoid use of jquery in new code

**Review by Prochnost55 — CHANGES_REQUESTED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 5 above)
Comment (Prochnost55) [reply webplugin/js/app/mck-sidebox-1.0.js:2963]: Ok

**Review by Prochnost55 — APPROVED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Comment by devashishmamgain** [comment]
@Ojashwa how ill this drop down value be shown in the dashboard conversation section? can you please add a screenshot of it.

**Comment by Ojashwa** [comment]
> @Ojashwa how ill this drop down value be shown in the dashboard conversation section? can you please add a screenshot of it.

image

**Pull request merged**